src/arraystring.rs

```diff
@@ -577,10 +472,15 @@ where
     #[inline]
     pub fn try_push_str(&mut self, string: impl AsRef<str>) -> Result<(), OutOfBounds> {
         trace!("Push str: {}", string.as_ref());
-        let new_end = string.as_ref().len().saturating_add(self.len().into());
-        is_inside_boundary(new_end, Self::capacity())?;
-        unsafe { self.push_str_unchecked(string) };
-        Ok(())
+        let str = string.as_ref().as_bytes();
+        if str.len() == 0 {
+            return Ok(());
+        }
+        is_inside_boundary(str.len() + self.len(), Self::capacity())?;
+        let this_len = self.len();
+        self.array[this_len..this_len + str.len()].copy_from_slice(str);
```

Comment:
This can technically panic. I'm ok removing that garantee, but I would rather not.

But if there isn't any better way of doing it, more power to this

Reply:
it shouldn't ever, both sizes are the same so it should be optimized out. let me add a compiler check so we can continue that guarantee. I'm all for no panics.

Reply:
I agree that it shouldn't ever, but in my experience the only way to ensure that is with get_unchecked, but if you can prove me wrong that would be awesome!

Reply:
indeed it is not optimized out...

Reply:
well you won't ever beleive where a panic was lol:
the intos of `is_inside_boundary` had an unoptimized one for some reason...

Reply:
Wow that is weird! Let's just remove the Into bounds then.

Reply:
and it wasn't able to optimize out the panic, even if i did an explicit check... crazy. maybe it wasn't inlining the function properly...

Reply:
> Wow that is weird! Let's just remove the Into bounds then.

it's done